src/main/java/com/twilio/appointmentreminders/controllers/AppointmentController.java

```diff
@@ -1,9 +1,135 @@
 package com.twilio.appointmentreminders.controllers;
 
+import com.twilio.appointmentreminders.models.Appointment;
+import com.twilio.appointmentreminders.models.AppointmentService;
+import com.twilio.appointmentreminders.util.AppointmentScheduler;
+import com.twilio.appointmentreminders.util.FieldValidator;
+import com.twilio.appointmentreminders.util.TimeZones;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.quartz.JobDetail;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import spark.ModelAndView;
 import spark.Route;
+import spark.TemplateViewRoute;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.quartz.JobBuilder.*;
+import static org.quartz.TriggerBuilder.newTrigger;
 
 public class AppointmentController {
-  public static Route sayHello = (request, response) -> {
-    return "hello functional stuff";
-  };
+    private AppointmentService service;
+    private Scheduler scheduler;
+
+    public AppointmentController(AppointmentService service, Scheduler scheduler) {
+        this.service = service;
+        this.scheduler = scheduler;
+    }
+
+    public TemplateViewRoute renderCreatePage = (request, response) -> {
+        Map map = new HashMap();
+
+        map.put("zones", timeZones());
+        return new ModelAndView(map, "new.mustache");
+    };
+
+    public TemplateViewRoute create = (request, response) -> {
+        FieldValidator validator = new FieldValidator(
+                new String[]{"name", "phoneNumber", "date", "delta", "timeZone"}
+        );
+
+        if (validator.valid(request)) {
+            try {
+                String name = request.queryParams("name");
+                String phoneNumber = request.queryParams("phoneNumber");
+                String date = request.queryParams("date");
+                int delta = Integer.parseInt(request.queryParams("delta"));
+                String timeZone = request.queryParams("timeZone");
+
+                DateTimeZone zone = DateTimeZone.forID(timeZone);
+                DateTimeZone zoneUTC = DateTimeZone.UTC;
+
+                DateTime dt;
+                DateTimeFormatter formatter = DateTimeFormat.forPattern("MM-dd-yyyy hh:mma");
+                formatter = formatter.withZone(zone);
+                dt = formatter.parseDateTime(date);
+                formatter = formatter.withZone(zoneUTC);
+                String dateUTC = dt.toString(formatter);
+
+                Appointment appointment = new Appointment(name, phoneNumber, delta, dateUTC, timeZone);
+                service.create(appointment);
+
+                scheduleJob(appointment);
+
+                response.redirect("/");
```

Comment:
Is the whole `try` block capable of throwing this exception? Ideally we want to wrap as little code as possible in a `try` `catch`.